lib/SILOptimizer/Transforms/SpeculativeDevirtualizer.cpp

```diff
@@ -47,6 +47,39 @@ static const int MaxNumSpeculativeTargets = 6;
 
 STATISTIC(NumTargetsPredicted, "Number of monomorphic functions predicted");
 
+/// We want to form a second edge to the given block, but we know
+/// that'll form a critical edge.
+static SILBasicBlock *breakCriticalEdge(TermInst *TI, unsigned SuccIndex) {
```

Comment:
There is already a helper that does this:

https://github.com/apple/swift/blob/master/include/swift/SILOptimizer/Utils/CFG.h#L74

Reply:
Thanks.  But why does it clone the terminator instead of just rewriting its successor?

Reply:
...this is actually a serious problem for this code because it invalidates the existing ApplySite.

Reply:
I am not sure. I imagine it was just an oversight (@atrick and @aschwaighofer may remember). I would be fine with changing the routine to not do that.

Reply:
It may date from an era when that was necessary, although the better fix then would still have been to change the instructions to allow successor modification.

I've added a commit which changes that function to rewrite successors in-place when it isn't dropping arguments.